Generate.py

```diff
@@ -378,13 +378,44 @@ def roll_linked_options(weights: dict) -> dict:
     return weights
 
 
+def compare_results(
+        yaml_value: Union[str, int, bool, dict, list],
+        trigger_value: Union[str, int, bool, dict, list],
+        comparator: str):
+    if yaml_value is None:
+        return False
+    if isinstance(yaml_value, str|bool|int) and isinstance(trigger_value, str|bool|int):
+        yaml_value = str(yaml_value).lower()
+        trigger_value = str(trigger_value).lower()
+    if comparator == "=":
+        return yaml_value == trigger_value
+    if comparator == "!=":
+        return yaml_value != trigger_value
+    try:
+        yaml_value = int(yaml_value)
+        trigger_value = int(trigger_value)
+    except ValueError:
```

Comment:
This doesn't catch TypeErrors from lists and dictionaries.

Reply:
Fixed to include TypeError